packages/luigis-box/src/FrontendApi/Resolver/Recommendation/RecommendationQuery.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\LuigisBoxBundle\FrontendApi\Resolver\Recommendation;
+
+use Overblog\GraphQLBundle\Definition\Argument;
+use Shopsys\FrameworkBundle\Model\Product\ProductElasticsearchProvider;
+use Shopsys\FrontendApiBundle\Model\Resolver\AbstractQuery;
+use Shopsys\LuigisBoxBundle\Component\LuigisBox\LuigisBoxClient;
+use Shopsys\LuigisBoxBundle\Model\Batch\LuigisBoxBatchLoadDataFactory;
+use Shopsys\LuigisBoxBundle\Model\Type\TypeInLuigisBoxEnum;
+use Shopsys\ProductFeed\LuigisBoxBundle\Model\FeedItem\LuigisBoxProductFeedItem;
+
+class RecommendationQuery extends AbstractQuery
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductElasticsearchProvider $productElasticsearchProvider
+     * @param \Shopsys\LuigisBoxBundle\Component\LuigisBox\LuigisBoxClient $luigisBoxClient
+     * @param \Shopsys\LuigisBoxBundle\Model\Batch\LuigisBoxBatchLoadDataFactory $luigisBoxBatchLoadDataFactory
+     */
+    public function __construct(
+        protected readonly ProductElasticsearchProvider $productElasticsearchProvider,
+        protected readonly LuigisBoxClient $luigisBoxClient,
+        protected readonly LuigisBoxBatchLoadDataFactory $luigisBoxBatchLoadDataFactory,
+    ) {
+    }
+
+    /**
+     * @param \Overblog\GraphQLBundle\Definition\Argument $argument
+     * @return \Shopsys\FrameworkBundle\Model\Product\Product[]
```

Comment:
I do not think `Product` instances are returned - the data are fetched from Elasticsearch, right?

Reply:
Changed